public/app/features/alerting/unified/utils/redux.ts

```diff
@@ -13,15 +15,21 @@ export interface AsyncRequestState<T> {
   requestId?: string;
 }
 
-export const initialAsyncRequestState: Pick<
-  AsyncRequestState<undefined>,
-  'loading' | 'dispatched' | 'result' | 'error'
-> = Object.freeze({
-  loading: false,
-  result: undefined,
-  error: undefined,
-  dispatched: false,
-});
+interface AsyncInitialRequestState<T> extends AsyncRequestState<T> {
+  result: T;
+}
+
+export const createInitialAsyncRequestState = <T>(
+  state: T
+): Pick<AsyncInitialRequestState<T>, 'loading' | 'dispatched' | 'result' | 'error'> =>
+  Object.freeze({
```

Comment:
This code was already in place, I changed it a bit to allow an initial state, instead of `undefined`. But yes, by doing a freeze, they make sure this initial state is not altered directly. When working with reducers, it's a rule to always create a new state from the previous one rather then changing the current one. `Object.freeze` makes sure that this particular object won't change.